FW/FreeRTOS/Task.cpp

```diff
@@ -0,0 +1,71 @@
+#include "Task.hpp"
+
+namespace FreeRTOS
+{
+    Task::Task(const char *name, const uint16_t stackSize, const UBaseType_t priority) {
+        for(uint32_t i = 0; i < configMAX_TASK_NAME_LEN; i++) {
+            m_name[i] = name[i];
+        }
```

Comment:
If name is less than ```configMAX_TASK_NAME_LEN``` this will read from outside it. If name is longer, m_name won't be null terminated. I suggest:

```c++
strncpy(m_name, name, configMAX_TASK_NAME_LEN);
// strncpy() does not guarantee null-termination.
m_name[configMAX_TASK_NAME_LEN - 1] = '\0';
```